components/authz-service/storage/postgres/migration/sql/68_add_status_to_projects.up.sql

```diff
@@ -0,0 +1,40 @@
+BEGIN;
+
+CREATE OR REPLACE FUNCTION
+  query_projects(_projects_filter TEXT[])
+  RETURNS setof json AS $$
+
+WITH
+staged AS (
+  SELECT r.project_id, count(r.id) AS count
+  FROM iam_staged_project_rules AS r
+  WHERE projects_match_for_rule(r.project_id, _projects_filter) AND r.deleted = false
+  GROUP BY r.project_id
+),
+applied AS (
+  SELECT r.project_id, count(r.id) AS count
+  FROM iam_project_rules AS r
+  WHERE projects_match_for_rule(r.project_id, _projects_filter)
+  GROUP BY r.project_id
+)
+SELECT json_build_object(
+  'id', p.id,
+  'name', p.name,
+  'type', p.type,
+  'status', 
+  CASE WHEN staged.count IS NULL AND applied.count IS NULL THEN 'no-rules'
+       WHEN staged.count IS NULL THEN 'applied'
+       ELSE 'edits-pending'
+  END
```

Comment:
The changes to `query_project` are all centered around adding this computed `status` field to the output object. All the work is done here in the DB, saving a lot of Go-based machinations.

Reply:
Not necessarily faster, but I believe it is more suitable here for two reasons:
1. We're already doing part of the work here in determining the result of `ListProjects`; it makes sense to do the rest of it here, too.
2. Go is rather verbose at times due to its lack of simple things common to many high-level languages (e.g., an `array-contains()` function). I believe the number of lines of code to generate this output in Go would be greater.